src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py

```diff
@@ -316,7 +338,24 @@ def minimize_testcase(self, target_path, arguments, input_path, output_path,
     Raises:
       TimeoutError: If the testcase minimization exceeds max_time.
     """
-    raise NotImplementedError
+    runner = _get_runner(target_path)
+    workdir = self._create_temp_dir('workdir')
+    args = [
+        f'--binary={target_path}',
+        f'--workdir={workdir}',
+        f'--minimize_crash={input_path}',
+        f'--num_runs={constants.NUM_RUNS_PER_MINIMIZATION}',
+        '--seed=1',
+    ]
+    result = runner.run_and_wait(additional_args=args, timeout=max_time)
+    if result.timed_out:
+      logs.log_error(
```

Comment:
We should be consistent here and follow libfuzzer's approach of raising a timeouterror isntead: https://github.com/google/clusterfuzz/blob/7325e7d2d7e4a30ce2a599cbb1f1e5cceac8643c/src/clusterfuzz/_internal/bot/fuzzers/libFuzzer/engine.py#L567

Reply:
Fixed.